src/tools/compiletest/src/runtest.rs

```diff
@@ -1399,7 +1399,12 @@ impl<'test> TestCx<'test> {
         let mut mir_dump_dir = self.config.build_test_suite_root.clone();
         debug!("input_file: {}", self.testpaths.file);
         mir_dump_dir.push(&self.testpaths.relative_dir);
-        mir_dump_dir.push(self.testpaths.file.file_stem().unwrap());
+        let name = self.testpaths.file.file_stem().unwrap();
+        if let Some(revision) = self.revision {
+            mir_dump_dir.push(format!("{name}#{revision}"));
+        } else {
+            mir_dump_dir.push(name);
+        }
```

Comment:
Problem: I believe this needs to go through same handling as https://github.com/rust-lang/rust/blob/b8005bff3248cfc6e327faf4fa631ac49bb49ba9/src/tools/compiletest/src/runtest.rs#L1858-L1861

Note the "safe_revision" bit. Maybe you can use https://github.com/rust-lang/rust/blob/b8005bff3248cfc6e327faf4fa631ac49bb49ba9/src/tools/compiletest/src/common.rs#L891-L901

Reply:
So do we even need this delete/create thing? That should be handled by our general test clean-up code.

Reply:
We used to because mir-opt didn't do per-revision test dir, but if we're creating per-revision test dir then no, we don't need to cleanup per-revision.

Reply:
Actually, let me double-check this

Reply:
> That should be handled by our general test clean-up code.

AFAICT there isn't such a "general test clean-up code" because compiletest uses input stamps for libtest "up-to-date" handling. The test binary is attempted to be removed cf. https://github.com/rust-lang/rust/blob/c8f94230282a8e8c1148f3e657f0199aad909228/src/tools/compiletest/src/runtest.rs#L1042-L1046 if test is successful.

However, e.g. `.mir` or `.{stderr,stdout}` are not removed on rerun for the "up-to-date" reasons, and AFAICT are simply overwritten.

Reply:
Makes sense to me! Thanks for looking into that.